ament_index_cpp/QUALITY_DECLARATION.md

```diff
@@ -121,7 +121,13 @@ A description of how coverage statistics are summarized from this page, can be f
 
 ### Performance [4.iv]
 
-`ament_index_cpp` does not conduct performance tests.
+An environment variable defines the prefix paths of such resource indices and the API has a time complexity of `O(n)` where `n` is the number of prefix paths.
+The time complexity to query information is either scaling lineraly with the number of resource types or with the number of resources per type (depending on which dimension is requested).
+If the content of a specific resource is retrieved the time complexity is linear to the size of the content as is the memory usage in that case since the content is returned to the caller.
+The performance of the implementation is defined by the performance of the underlying filesystem functions and the implemented logic doesn't add any significant overhead.
```

Comment:
@dirk-thomas meta: while I agree the logic overhead is likely negligible compared to that of IO, the performance of `ament_index` queries has more to do with the way the index is structured than with the code itself.

Reply:
That is why I tried to describe the complexity of the different kind of queries available.

Arguably if there would be a more efficient way the resource index could be structured to make queries more efficient (which still satisfies the requirements) that might be something better discussed on the design document (https://github.com/ament/ament_cmake/blob/master/ament_cmake_core/doc/resource_index.md).

Reply:
> Arguably if there would be a more efficient way the resource index could be structured to make queries more efficient

Sure, my point being that to say `The performance of the implementation is defined by the performance of the underlying filesystem functions` is not entirely correct.

Reply:
Can you elaborate why?

Reply:
Re-reading this thread, I get the impression that we're using the term `performance` for different things (algorithmic complexity vs. runtime cost). Since you do mention time-complexity, perhaps rephrasing as:

```suggestion
The runtime cost of the implementation is dominated by the runtime cost of the underlying filesystem API, and the implemented logic doesn't add any significant overhead.
```

would constrain interpretation. WDYT?

Reply:
Sounds good to me - applied in both locations in d0dbd707f6a25ef106f1a3265e13f86b7db1cfb8.